In [63]:
import pandas as pd

In [64]:
df = pd.read_csv(
    "Database export - Clothing distribuiton EMEA - 2019-2021.csv", encoding="latin1"
)


In [65]:
df.drop("Unnamed: 0", axis=1, inplace=True)

In [66]:
df.head()

Company Name ISO Country Code               Country  \
0  ITX HELLAS SINGLE MEMBER S.A.               GR                 Grèce   
1                        RANDEVU               RU  Fédération de Russie   
2           ZEEMAN TEXTIELSUPERS               FR                France   
3               ANSWEAR.COM S.A.               PL               Pologne   
4                    REGENT GOLD               RU  Fédération de Russie   

                                         City  \
0                                      ATHENS   
1                                      MOSCOW   
2                                       PARIS   
3                                      KRAKOW   
4  AGALATOVSKOE SELSKOE POSELENIE, D. SKOTNOE   

                                          Address  NACE Rev. 2 Code principal  \
0                                             NaN                        4771   
1             UL. SKAKOVAYA D. 17, STR. 2, KOM. 2                        4772   
2                                             NaN                        4771   
3                                             NaN                        4771   
4  UL. SAVUSHKINA D. 126, LIT. A, POMESHCH. 133-N                        4777   

                 NACE Principal Rev. 2, description   \
0  Commerce de détail d'habillement en magasin sp...   
1  Commerce de détail de chaussures et d'articles...   
2  Commerce de détail d'habillement en magasin sp...   
3  Commerce de détail d'habillement en magasin sp...   
4  Commerce de détail d'articles d'horlogerie et ...   

                                              Sector  \
0          Engaged in the retail of fashion clothing   
1  Engaged in the wholesale trade and marketing o...   
2  Engaged in the retail sale of clothing and out...   
3  Digital platform for the sale of branded cloth...   
4     Operates as other miscellaneous store retailer   

                 Main activity  \
0                       Retail   
1                    Wholesale   
2                       Retail   
3  Wholesale; Retail; Services   
4                       Retail   

                    Description in Original language  ...  \
0                                                NaN  ...   
1                     ?????????? ????? ? ???????????  ...   
2  Négoce au détail ou en gros de vêtements, acce...  ...   
3                                                NaN  ...   
4                      ???????? ?????????? ?????????  ...   

                                   Website  \
0        www.zara.com\nwww.fashionwear1.gr   
1  www.rendez-vous.ru\nwww.rest-randevu.ru   
2                           www.zeeman.com   
3                          www.answear.com   
4                           www.zolotoy.ru   

  Operating result [=EBIT]\nmEUR 2021 Operating result [=EBIT]\nmEUR 2020  \
0                                  31                                  17   
1                                  15                                  14   
2                                   4                                   3   
3                                   8                                   5   
4                                  22                                  16   

  Operating result [=EBIT]\nmEUR 2019 Turnover\nmEUR 2021  \
0                                  22                 459   
1                                   1                 306   
2                                   4                 164   
3                                   1                 149   
4                                   5                 141   

   Turnover\nmEUR 2020  Turnover\nmEUR 2019  Availability of financial data  \
0                  352                  300                             YES   
1                  230                  337                             YES   
2                  137                  147                             YES   
3                   90                   73                             YES   
4                  103                 

In [67]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 490 entries, 0 to 489
Data columns (total 21 columns):
 #   Column                               Non-Null Count  Dtype 
---  ------                               --------------  ----- 
 0   Company Name                         490 non-null    object
 1   ISO Country Code                     490 non-null    object
 2   Country                              490 non-null    object
 3   City                                 490 non-null    object
 4   Address                              218 non-null    object
 5   NACE Rev. 2 Code principal           490 non-null    int64 
 6   NACE Principal Rev. 2, description   490 non-null    object
 7   Sector                               322 non-null    object
 8   Main activity                        322 non-null    object
 9   Description in Original language     339 non-null    object
 10  Description in English               346 non-null    object
 11  Website                              436 non-

In [68]:
# ✅ 1. Clean Column Names
df.columns = [col.replace("\n", " ").strip() for col in df.columns]


In [69]:
# ✅ 2. Convert Financial Columns to Numeric
financial_cols = [
    "Operating result [=EBIT] mEUR 2021",
    "Operating result [=EBIT] mEUR 2020",
    "Operating result [=EBIT] mEUR 2019",
    "Turnover mEUR 2021",
    "Turnover mEUR 2020",
    "Turnover mEUR 2019",
]
df[financial_cols] = df[financial_cols].apply(pd.to_numeric, errors="coerce")

In [70]:
# ✅ 3. Handle Missing Values
df["Website"] = df["Website"].fillna("No Website")  # Replace missing websites
df.dropna(subset=["Description in English"], inplace=True)

In [71]:
# ✅ 4. Extract First Website from Multi-Entry Fields
df["Website"] = df["Website"].apply(
    lambda x: x.split("\n")[0] if isinstance(x, str) else x
)


In [72]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 346 entries, 0 to 488
Data columns (total 21 columns):
 #   Column                              Non-Null Count  Dtype  
---  ------                              --------------  -----  
 0   Company Name                        346 non-null    object 
 1   ISO Country Code                    346 non-null    object 
 2   Country                             346 non-null    object 
 3   City                                346 non-null    object 
 4   Address                             141 non-null    object 
 5   NACE Rev. 2 Code principal          346 non-null    int64  
 6   NACE Principal Rev. 2, description  346 non-null    object 
 7   Sector                              322 non-null    object 
 8   Main activity                       322 non-null    object 
 9   Description in Original language    232 non-null    object 
 10  Description in English              346 non-null    object 
 11  Website                             346 non-null  

In [73]:
df.drop(["Description in Original language"], axis=1, inplace=True)

In [74]:
df

Company Name ISO Country Code               Country  \
0    ITX HELLAS SINGLE MEMBER S.A.               GR                 Grèce   
1                          RANDEVU               RU  Fédération de Russie   
2             ZEEMAN TEXTIELSUPERS               FR                France   
3                 ANSWEAR.COM S.A.               PL               Pologne   
4                      REGENT GOLD               RU  Fédération de Russie   
..                             ...              ...                   ...   
482                BIJOUX BOUTIQUE               FR                France   
483          PRO RETAIL FINLAND OY               FI              Finlande   
484                  BIONDI S.R.L.               IT                Italie   
487                      ZETEK SRL               IT                Italie   
488                   WADEL S.R.L.               IT                Italie   

                                           City  \
0                                        ATHENS   
1                                        MOSCOW   
2                                         PARIS   
3                                        KRAKOW   
4    AGALATOVSKOE SELSKOE POSELENIE, D. SKOTNOE   
..                                          ...   
482                                  STRASBOURG   
483                                       ESPOO   
484                                    VERCELLI   
487                                      CERVIA   
488                                    CAGLIARI   

                                            Address  \
0                                               NaN   
1               UL. SKAKOVAYA D. 17, STR. 2, KOM. 2   
2                                               NaN   
3                                               NaN   
4    UL. SAVUSHKINA D. 126, LIT. A, POMESHCH. 133-N   
..                                              ...   
482                                             NaN   
483         C/O KOTKAN TILIPALVELU OY KIRKKOKATU 18   
484                               CORSO LIBERTA' 41   
487                             VIALE ROMAGNA 69/71   
488                          VIA DEL RISORGIMENTO 5   

     NACE Rev. 2 Code principal  \
0                          4771   
1                          4772   
2                          4771   
3                          4771   
4                          4777   
..                          ...   
482                        4777   
483                        4771   
484                        4777   
487                        4771   
488                        4772   

                    NACE Principal Rev. 2, description  \
0    Commerce de détail d'habillement en magasin sp...   
1    Commerce de détail de chaussures et d'articles...   
2    Commerce de détail d'habillement en magasin sp...   
3    Commerce de détail d'habillement en magasin sp...   
4    Commerce de détail d'articles d'horlogerie et ...   
..                                                 ...   
482  Commerce de détail d'articles d'horlogerie et ...   
483  Commerce de détail d'habillement en magasin sp...   
484  Commerce de détail d'articles d'horlogerie et ...   
487  Commerce de détail d'habillement en magasin sp...   
488  Commerce de détail de chaussures et d'articles...   

                                                Sector  \
0            Engaged in the retail of fashion clothing   
1    Engaged in the wholesale trade and marketing o...   
2    Engaged in the retail sale of clothing and out...   
3    Digital platform for the sale of branded cloth...   
4       Operates as other miscellaneous store retailer   
..                                                 ...   
482  Operates as a retailer of miscellaneous shoppi...   
483                                                NaN   
484  Primarily engaged in the wholesale trade of je...   
487  Engaged in the manufacture and retail sale of ...   
488  Primarily engaged in the retail sale of footwe...   

        

In [75]:
df["NACE Rev. 2 Code principal"].unique()

array([4771, 4772, 4777, 4751, 4641, 4511, 2611])

| NACE Code | Description                                                                 |
|-----------|-----------------------------------------------------------------------------|
| 4771      | Retail sale of clothing in specialized stores ✅ (Relevant for luxury filtering) |
| 4772      | Retail sale of footwear and leather goods ✅ (Luxury possible, needs filtering) |
| 4777      | Retail sale of watches and jewelry in specialized stores ✅ (Luxury brands possible) |
| 4751      | Retail sale of textiles (e.g., fabrics, not clothing) ❌ (Likely irrelevant) |
| 4641      | Wholesale of textiles ❌ (Not a retailer, likely irrelevant) |
| 4511      | Sale of motor vehicles ❌ (Not relevant to clothing/luxury) |
| 2611      | Manufacture of electronic components ❌ (Completely irrelevant) |
| 4764      | Retail sale of sports goods ❌ (Sports-focused, unlikely to be luxury) |
| 4615      | Wholesale of furniture, carpets, lighting equipment ❌ (Irrelevant) |

In [76]:
# relevant NACE codes
codes = [4771, 4772, 4777]

In [77]:
# Keep only rows with relevant NACE codes
df = df[df["NACE Rev. 2 Code principal"].isin(codes)]


In [79]:
df

Company Name ISO Country Code               Country  \
0    ITX HELLAS SINGLE MEMBER S.A.               GR                 Grèce   
1                          RANDEVU               RU  Fédération de Russie   
2             ZEEMAN TEXTIELSUPERS               FR                France   
3                 ANSWEAR.COM S.A.               PL               Pologne   
4                      REGENT GOLD               RU  Fédération de Russie   
..                             ...              ...                   ...   
326                BIJOUX BOUTIQUE               FR                France   
327          PRO RETAIL FINLAND OY               FI              Finlande   
328                  BIONDI S.R.L.               IT                Italie   
329                      ZETEK SRL               IT                Italie   
330                   WADEL S.R.L.               IT                Italie   

                                           City  \
0                                        ATHENS   
1                                        MOSCOW   
2                                         PARIS   
3                                        KRAKOW   
4    AGALATOVSKOE SELSKOE POSELENIE, D. SKOTNOE   
..                                          ...   
326                                  STRASBOURG   
327                                       ESPOO   
328                                    VERCELLI   
329                                      CERVIA   
330                                    CAGLIARI   

                                            Address  \
0                                               NaN   
1               UL. SKAKOVAYA D. 17, STR. 2, KOM. 2   
2                                               NaN   
3                                               NaN   
4    UL. SAVUSHKINA D. 126, LIT. A, POMESHCH. 133-N   
..                                              ...   
326                                             NaN   
327         C/O KOTKAN TILIPALVELU OY KIRKKOKATU 18   
328                               CORSO LIBERTA' 41   
329                             VIALE ROMAGNA 69/71   
330                          VIA DEL RISORGIMENTO 5   

     NACE Rev. 2 Code principal  \
0                          4771   
1                          4772   
2                          4771   
3                          4771   
4                          4777   
..                          ...   
326                        4777   
327                        4771   
328                        4777   
329                        4771   
330                        4772   

                    NACE Principal Rev. 2, description  \
0    Commerce de détail d'habillement en magasin sp...   
1    Commerce de détail de chaussures et d'articles...   
2    Commerce de détail d'habillement en magasin sp...   
3    Commerce de détail d'habillement en magasin sp...   
4    Commerce de détail d'articles d'horlogerie et ...   
..                                                 ...   
326  Commerce de détail d'articles d'horlogerie et ...   
327  Commerce de détail d'habillement en magasin sp...   
328  Commerce de détail d'articles d'horlogerie et ...   
329  Commerce de détail d'habillement en magasin sp...   
330  Commerce de détail de chaussures et d'articles...   

                                                Sector  \
0            Engaged in the retail of fashion clothing   
1    Engaged in the wholesale trade and marketing o...   
2    Engaged in the retail sale of clothing and out...   
3    Digital platform for the sale of branded cloth...   
4       Operates as other miscellaneous store retailer   
..                                                 ...   
326  Operates as a retailer of miscellaneous shoppi...   
327                                                NaN   
328  Primarily engaged in the wholesale trade of je...   
329  Engaged in the manufacture and retail sale of ...   
330  Primarily engaged in the retail sale of footwe...   

        

In [80]:
df["Main activity"].unique()

array(['Retail', 'Wholesale', 'Wholesale; Retail; Services',
       'Manufacturing; Retail', 'Manufacturing', 'Services',
       'Retail; Services', nan, 'Wholesale; Retail',
       'Manufacturing; Wholesale', 'Manufacturing; Retail; Services',
       'Wholesale; Services'], dtype=object)

In [82]:
relevant_main_activity = [
    "Retail",
    "Wholesale; Retail; Services",
    "Manufacturing; Retail",
    "Retail; Services",
    "Wholesale; Retail",
    "Manufacturing; Retail; Services",
]

df = df[df["Main activity"].isin(relevant_main_activity)]

In [83]:
df

Company Name ISO Country Code               Country  \
0    ITX HELLAS SINGLE MEMBER S.A.               GR                 Grèce   
2             ZEEMAN TEXTIELSUPERS               FR                France   
3                 ANSWEAR.COM S.A.               PL               Pologne   
4                      REGENT GOLD               RU  Fédération de Russie   
6     NILSON GROUP AB FILIAL NORGE               NO               Norvège   
..                             ...              ...                   ...   
322                  MARMON SPORTS               FR                France   
325             MODA CENTER S.R.L.               IT                Italie   
326                BIJOUX BOUTIQUE               FR                France   
329                      ZETEK SRL               IT                Italie   
330                   WADEL S.R.L.               IT                Italie   

                                           City  \
0                                        ATHENS   
2                                         PARIS   
3                                        KRAKOW   
4    AGALATOVSKOE SELSKOE POSELENIE, D. SKOTNOE   
6                                          OSLO   
..                                          ...   
322                               AUBERVILLIERS   
325                                  TREVIGNANO   
326                                  STRASBOURG   
329                                      CERVIA   
330                                    CAGLIARI   

                                            Address  \
0                                               NaN   
2                                               NaN   
3                                               NaN   
4    UL. SAVUSHKINA D. 126, LIT. A, POMESHCH. 133-N   
6                                               NaN   
..                                              ...   
322                                             NaN   
325                                  VIA TREVISO 71   
326                                             NaN   
329                             VIALE ROMAGNA 69/71   
330                          VIA DEL RISORGIMENTO 5   

     NACE Rev. 2 Code principal  \
0                          4771   
2                          4771   
3                          4771   
4                          4777   
6                          4772   
..                          ...   
322                        4771   
325                        4771   
326                        4777   
329                        4771   
330                        4772   

                    NACE Principal Rev. 2, description  \
0    Commerce de détail d'habillement en magasin sp...   
2    Commerce de détail d'habillement en magasin sp...   
3    Commerce de détail d'habillement en magasin sp...   
4    Commerce de détail d'articles d'horlogerie et ...   
6    Commerce de détail de chaussures et d'articles...   
..                                                 ...   
322  Commerce de détail d'habillement en magasin sp...   
325  Commerce de détail d'habillement en magasin sp...   
326  Commerce de détail d'articles d'horlogerie et ...   
329  Commerce de détail d'habillement en magasin sp...   
330  Commerce de détail de chaussures et d'articles...   

                                                Sector  \
0            Engaged in the retail of fashion clothing   
2    Engaged in the retail sale of clothing and out...   
3    Digital platform for the sale of branded cloth...   
4       Operates as other miscellaneous store retailer   
6    Engaged in the operation of a chain of shoe st...   
..                                                 ...   
322  Engaged in the retail trade of sporting goods ...   
325  Engaged in the retail trade of wearing apparel...   
326  Operates as a retailer of miscellaneous shoppi...   
329  Engaged in the manufacture and retail sale of ...   
330  Primarily engaged in the retail sale of footwe...   

        

In [84]:
df.reset_index(drop=True, inplace=True)


In [ ]:
df.to_csv